src/box/xlog.c

```diff
@@ -1526,7 +1526,7 @@ xlog_close(struct xlog *l, bool reuse_fd)
 
 /* {{{ struct xlog_cursor */
 
-#define XLOG_READ_AHEAD		(1 << 14)
+#define XLOG_READ_BUF_SIZE	(8 * 1024 * 1024)
```

Comment:
xlog cursor is used not only for recovery, but also for replication so raising the read ahead size up to 8 MB may lead to memory consumption increase by up to 256 MB per instance with registered replicas and even more with anonymous replicas. I'm not sure we can afford that. Maybe, increase the buffer size only for recovery? @alyapunov WDYT?

Reply:
Fixed by a dynamically-sized read buffer.